src/pltt_buffer.c

```diff
@@ -0,0 +1,803 @@
+    r0->unk14 = r6;
+    r4 = r0->unk20;
+    r7 = r0->unkC;
+    r5 = gUnk_08B714AC[r0->unk2];
+    if (r4 < 0)
+    {
+        r5(&sb[0x7E0], &sb[0x7E0], r7, r6);
+    }
+    if (r4 & (0x80 << 0x17))
+    {
+        r5(&sb[0x7C0], &sb[0x7C0], r7, r6);
+    }
+    if (r4 & (0x80 << 0x16))
+    {
+        r5(&sb[0x7A0], &sb[0x7A0], r7, r6);
+    }
+    if (r4 & (0x80 << 0x15))
+    {
+        r5(&sb[0x780], &sb[0x780], r7, r6);
+    }
+    if (r4 & (0x80 << 0x14))
+    {
+        r5(&sb[0x760], &sb[0x760], r7, r6);
+    }
+    if (r4 & (0x80 << 0x13))
+    {
+        r5(&sb[0x740], &sb[0x740], r7, r6);
+    }
+    if (r4 & (0x80 << 0x12))
+    {
+        r5(&sb[0x720], &sb[0x720], r7, r6);
+    }
+    if (r4 & (0x80 << 0x11))
+    {
+        r5(&sb[0x700], &sb[0x700], r7, r6);
+    }
+    if (r4 & (0x80 << 0x10))
+    {
+        r5(&sb[0x6E0], &sb[0x6E0], r7, r6);
+    }
+    if (r4 & (0x80 << 0xF))
+    {
+        r5(&sb[0x6C0], &sb[0x6C0], r7, r6);
+    }
+    if (r4 & (0x80 << 0xE))
+    {
+        r5(&sb[0x6A0], &sb[0x6A0], r7, r6);
+    }
+    if (r4 & (0x80 << 0xD))
+    {
+        r5(&sb[0x680], &sb[0x680], r7, r6);
+    }
+    if (r4 & (0x80 << 0xC))
+    {
+        r5(&sb[0x660], &sb[0x660], r7, r6);
+    }
+    if (r4 & (0x80 << 0xB))
+    {
+        r5(&sb[0x640], &sb[0x640], r7, r6);
+    }
+    if (r4 & (0x80 << 0xA))
+    {
+        r5(&sb[0x620], &sb[0x620], r7, r6);
+    }
+    if (r4 & (0x80 << 0x9))
+    {
+        r5(&sb[0x600], &sb[0x600], r7, r6);
+    }
+    if (r4 & (0x80 << 0x8))
+    {
+        r5(&sb[0x5E0], &sb[0x5E0], r7, r6);
+    }
+    if (r4 & (0x80 << 0x7))
+    {
+        r5(&sb[0x5C0], &sb[0x5C0], r7, r6);
+    }
+    if (r4 & (0x80 << 0x6))
+    {
+        r5(&sb[0x5A0], &sb[0x5A0], r7, r6);
+    }
+    if (r4 & (0x80 << 0x5))
+    {
+        r5(&sb[0x580], &sb[0x580], r7, r6);
+    }
+    if (r4 & (0x80 << 0x4))
+    {
+        r5(&sb[0x560], &sb[0x560], r7, r6);
+    }
+    if (r4 & (0x80 << 0x3))
+    {
+        r5(&sb[0x540], &sb[0x540], r7, r6);
+    }
+    if (r4 & (0x80 << 0x2))
+    {
+        r5(&sb[0x520], &sb[0x520], r7, r6);
+    }
+    if (r4 & (0x80 << 0x1))
+    {
+        r5(&sb[0x500], &sb[0x500], r7, r6);
+    }
+    if (r4 & 0x80)
+    {
+        r5(&sb[0x4E0], &sb[0x4E0], r7, r6);
+    }
+    if (r4 & 0x40)
+    {
+        r5(&sb[0x4C0], &sb[0x4C0], r7, r6);
+    }
+    if (r4 & 0x20)
+    {
+        r5(&sb[0x4A0], &sb[0x4A0], r7, r6);
+    }
+    if (r4 & 0x10)
+    {
+        r5(&sb[0x480], &sb[0x480], r7, r6);
+    }
+    if (r4 & 0x8)
+    {
+        r5(&sb[0x460], &sb[0x460], r7, r6);
+    }
+    if (r4 & 0x4)
+    {
+        r5(&sb[0x440], &sb[0x440], r7, r6);
+    }
+    if (r4 & 0x2)
+    {
+        r5(&sb[0x420], &sb[0x420], r7, r6);
+    }
+    if (r4 & 0x1)
+    {
+        r5(&sb[0x400], &sb[0x400], r7, r6);
+    }
+
+    if (r0->unk8 == r0->unk6)
+    {
+        r0->unk0 = 3;
+    }
+    if (r0->unk0 == 4)
+    {
+        r0->unkA -= 1;
+        if (1) // ???
```

Comment:
If you can tell me what happens in the assembly, I may be able to resolve this part.